elements/lisk-p2p/test/integration/blacklist_whitelist_fixed_peers.ts

```diff
@@ -84,45 +74,33 @@ describe('Blacklisted/fixed/whitelisted peers', () => {
 		];
 
 		beforeEach(async () => {
-			p2pNodeList = [...new Array(NETWORK_PEER_COUNT).keys()].map(index => {
-				// Each node will have the previous node in the sequence as a seed peer except the first node.
-				const seedPeers = [
-					{
-						ipAddress: '127.0.0.' + (((index + 1) % NETWORK_PEER_COUNT) + 10),
-						wsPort: NETWORK_START_PORT + ((index + 1) % NETWORK_PEER_COUNT),
-					},
-				];
-				const nodePort = NETWORK_START_PORT + index;
-				return new P2P({
-					hostIp: '127.0.0.' + (index + 10),
-					connectTimeout: 100,
-					ackTimeout: 200,
-					blacklistedPeers: blacklistedPeers,
-					seedPeers,
-					fixedPeers: blacklistedPeers,
-					whitelistedPeers: blacklistedPeers,
-					previousPeers: previousPeersBlacklisted,
-					wsEngine: 'ws',
-					populatorInterval: POPULATOR_INTERVAL_WITH_LIMIT,
-					maxOutboundConnections: FIVE_CONNECTIONS,
-					maxInboundConnections: FIVE_CONNECTIONS,
-					nodeInfo: {
-						wsPort: nodePort,
-						nethash:
-							'da3ed6a45429278bac2666961289ca17ad86595d33b31037615d4b8e8f158bba',
-						version: '1.0.1',
-						protocolVersion: '1.0.1',
-						minVersion: '1.0.0',
-						os: platform(),
-						height: 0,
-						broadhash:
-							'2768b267ae621a9ed3b3034e2e8a1bed40895c621bbb1bbd613d92b9d24e54b5',
-						nonce: `O2wTkjqplHII${nodePort}`,
-					},
-				});
+			const customConfig = (index: number) => ({
```

Comment:
We shouldn't pass any callback functions to the `createNetwork`; that makes the code difficult to reason about. It forces the developer to dig into the boilerplate logic and keep a lot of things in their mind at the same time which distracts them from the important aspects of the test scenario.